### ISSUE TITLE
Library crashes when compiled to WASM with `atomics` flag enabled

### ISSUE THREAD

**Issue by drewhk** [mrDIMAS/tinyaudio#10]
When compiled to WASM with `atomics` flag (which flutter rust bridge enables) the library crashes due to 
```
.cargo\registry\src\index.crates.io-6f17d22bba15001f\tinyaudio-0.1.4\src\web.rs:57:14:
called `Result::unwrap()` on an `Err` value: JsValue(TypeError: Failed to execute 'copyToChannel' on 'AudioBuffer': The provided Float32Array value must not be shared.
TypeError: Failed to execute 'copyToChannel' on 'AudioBuffer': The provided Float32Array value must not be shared.
```
This error is exactly the same as documented in `cpal`: https://github.com/RustAudio/cpal/issues/656
Fixed here: https://github.com/RustAudio/cpal/pull/837/files

Basically with this flag the generated WASM code uses `SharedArrayBuffer`, but the `copyToChannel` only allows `Float32Array`. 
See also the related ticket on `wasm-bindgen`: https://github.com/rustwasm/wasm-bindgen/issues/3772

**Comment by mrDIMAS** [comment]
Cool, I'll submit a new version on crates.io then.

**Comment by mrDIMAS** [comment]
I've published `0.2.0`, closing this as completed.

**Comment by drewhk** [comment]
Many thanks! I owe you a beverage :D
